packages/react/src/List/ListItem.tsx

```diff
@@ -0,0 +1,40 @@
+import React, {type Ref, useContext} from 'react'
+import clsx from 'clsx'
+import {CheckIcon, DashIcon} from '@primer/octicons-react'
+import type {BaseProps} from '../component-helpers'
+import {ListContext} from './UnorderedList'
+
+import styles from './List.module.css'
+
+export type ListItemProps = BaseProps<HTMLElement> & {
+  /**
+   * React.ReactNode and React.ReactNode[] are valid children.
+   */
+  children?: React.ReactNode | React.ReactNode[]
+  /**
+   * The ref object to be attached to the list item.
+   */
+  ref?: Ref<HTMLLIElement>
+}
+
+export const ListItem = ({className, children, ...props}: ListItemProps) => {
```

Comment:
What's the context for having this as a separate component, rather than a namespaced member? I can't think of a use case for having this outside of List, so feels inconsistent with our other API's.

Reply:
If the reason is that it's being used across UnorderedList and OrderedList, I'd recommend importing it into those components as a dependency and then namespacing it. E.g. `List.Item`

Reply:
So I would change it to UnorderedList.Item. That works.

Reply:
In next push